### PR TITLE
:sparkles:Implement mobile nav and make navbar responsive for all device + fix job card and make it responsive (Fix #29 and #38)

### LINKED ISSUE #29: 🛠️ [UPDATE]: improve responsiveness of the landing page & mobile nav menu
The landing page needs to be responsive and we need to have a mobile nav menu, desktop navbar and mobile navbar can be kept as sperated components to avoid complexity. 

- Here is some references for the mobile nav menu: 
image

The navmenu can be implemented using shacn's sheet component
https://ui.shadcn.com/docs/components/sheet

### PR DESCRIPTION
**Pull request by NawinKumarSharma**
This pr fix #29
#38 
Before change:

[Screencast from 2024-08-06 16-19-41.webm](https://github.com/user-attachments/assets/9b523ff0-8bab-463e-8c1e-69b36cc6aaf2)

After change:

[Screencast from 2024-08-06 15-48-12.webm](https://github.com/user-attachments/assets/dc05ae58-6e49-4dac-a560-d5fc2578fc08)

### CODE REVIEW

**Comment by SwetanshuSingh** [comment]
@NawinKumarSharma heading, and subheadings are not following the correct text-size. The name 100xJobs got so small and it is the main title on the landing page. No proper alignment between ui elements also. Please, fix the addressed issues

**Comment by NawinKumarSharma** [comment]
@SwetanshuSingh  thanks for reviewing . i have come up with the change
[Screencast from 2024-08-06 18-21-08.webm](https://github.com/user-attachments/assets/db0ca157-bfe1-41ad-99d2-1ec2759b4409)

**Comment by Pheewww** [comment]
> @SwetanshuSingh thanks for reviewing . i have come up with the change. [Screencast from 2024-08-06 18-21-08.webm](https://github.com/user-attachments/assets/480d6c90-04b7-4576-9b47-3da4c689a79f)

this one is much better, ur nav is also cool.

Though the job card are still not good. maybe you can hide the sidebar as it will have filter which is not always needed on screen, then the jobs can be made cleaner. at present, they are like rectangle boxes in half the screen.

File: components/Navbar/MobileNav.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}: MobileLinkProps) {
+  const router = useRouter();
+  return (
+    <Link
+      href={href}
+      onClick={() => {
+        router.push(href.toString());
+        onOpenChange?.(false);
+      }}
+      className={`text-black hover:text-green-600 transition duration-300 mb-5 ${className}`}
```
Comment (VineeTagarwaL-code) [inline components/Navbar/MobileNav.tsx:116]: why are you not using cn() here, makes the code more easily readable and clean + editable

File: data/NavItems.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const NavItems = [
```
Comment (VineeTagarwaL-code) [inline data/NavItems.tsx:1]: rather than data rename the folder to constants, will give much more better idea

**Review by VineeTagarwaL-code — CHANGES_REQUESTED** [review]
Everything feels good, before merging i would love to see these changes made Great work LGTM !

File: data/NavItems.tsx (hunk 2 above)
Comment (NawinKumarSharma) [reply data/NavItems.tsx]: @VineeTagarwaL-code , thanks for reviewing. Done with requested changes.

**Review by VineeTagarwaL-code — APPROVED** [review]

**Pull request merged**